napalm_yang/parser.py

```diff
@@ -25,29 +25,27 @@ def __init__(self, model, device=None, profile=None, is_config=None,
         self.keys = keys or {"parent_key": None}
         self.extra_vars = extra_vars or {}
 
+        if self.mapping:
+            self.parser = helpers.get_parser(self.mapping.get("metadata", {}).get("parser", ''))
+
         if self.mapping and device:
-            device_config = self._execute_methods(device,
+            device_output = self._execute_methods(device,
                                                   self.mapping["metadata"].get("execute", []))
 
         else:
-            device_config = []
-
-        native = native or []
+            device_output = []
 
-        self.native = []
+        self.native = native or []
 
-        for n in native + device_config:
-            self.native.append(n.replace("\r", ""))  # Parsing will be easier
+        if hasattr(self, "parser"):
+            self.native = self.parser.init_native(self.native, device_output)
```

Comment:
I think it makes more sense if we do this inside the `parse` method, right before we call `self._parse`. That way you don't need to check if `self` has a parser.